lib/brains/tests/base.py

```diff
@@ -1,18 +1,83 @@
-from lib.brains import client
+import mock
+
 from lib.brains.errors import MockError
 from solitude.base import APITest
 
 
+class BraintreeMock(mock.Mock):
+
+    def __getattr__(self, name):
+        """
+        Record that a mock method was registered so we can check it
+        was called.
+        """
+        if not name.startswith('_'):
+            setattr(self, '_registered', [])
+            self._registered.append(name)
+        return super(BraintreeMock, self).__getattr__(name)
+
+    def tolist(self, *args, **kw):
+        """
+        DRF can get into a infinite loop with Mock on this method. If you
+        override the mock so it returns a value it works.
+
+        This is a hack of the worst kind.
+        """
+        raise MockError(
+            'Attempt to serialise a Mock without the result being overridden.')
+
+
 class BraintreeTest(APITest):
 
+    """
+    A standard APITest with the ability to provide Braintree mocks.
+
+    Add in dict of gateways to override onto the class, for example:
+
+            gateways = {'client': ClientTokenGateway}
+
+    Access the mock like this:
+
+            self.mocks['client'].generate.return_value = 'a-sample-token'
+
+    This class will throw a MockError if a result is not defined or if a
+    result was defined but never called.
+    """
+    gateways = {}
+
     def setUp(self):
-        self.set_mocks = client.set_mocks
         super(BraintreeTest, self).setUp()
 
-    def tearDown(self):
-        # You defined some mock calls for braintree, but didn't use them all.
-        if client.mocks:
-            raise MockError('Unused braintree client mock exists: {0}'
-                            .format(client.mocks))
-        self.set_mocks()
-        super(BraintreeTest, self).tearDown()
+        self.classes = {}
+        self.mocks = {}
+
+        for key, gateway in self.gateways.items():
+            name = gateway.__name__
```

Comment:
Does gateway have to be an object? I guess that forces each test to import it thus avoiding typos? Seems like ``{'client': 'ClientTokenGateway'}`` could be enough though.